src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.spring.security;
+
+import java.util.Optional;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email)
+            throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        UserBuilder builder;
+        if (userOptional.isPresent()) {
+            builder = org.springframework.security.core.userdetails.User.withUsername(email);
+            builder.password(userOptional.get().getPassword());
+            builder.roles(userOptional.get().getRoles()
+                    .stream()
+                    .map(x -> x.getRoleName().name())
+                    .toArray(String[]::new));
+            return builder.build();
+        }
+        throw new UsernameNotFoundException("User not found");
```

Comment:
better add email to the message
throw new UsernameNotFoundException("User with email: " + email + " not found");